submarine-workbench/submarine-server/src/main/java/org/apache/submarine/database/service/ProjectService.java

```diff
@@ -0,0 +1,158 @@
+                                     int pageSize) throws Exception {
+    LOG.info("queryPageList owner:{}, column:{}, order:{}, pageNo:{}, pageSize:{}",
+        userName, column, order, pageNo, pageSize);
+
+    List<Project> list = null;
+    try (SqlSession sqlSession = MyBatisUtil.getSqlSession()) {
+      ProjectMapper projectMapper = sqlSession.getMapper(ProjectMapper.class);
+      Map<String, Object> where = new HashMap<>();
+      where.put("userName", userName);
+      where.put("column", column);
+      where.put("order", order);
+      list = projectMapper.selectAll(where, new RowBounds(pageNo, pageSize));
+
+      ProjectFilesMapper projectFilesMapper = sqlSession.getMapper(ProjectFilesMapper.class);
+      // Query from project_files table, And set to Project Object
+      for (Project project : list) {
+        Map<String, Object> whereMember = new HashMap<>();
+        whereMember.put("projectId", project.getId());
+        List<ProjectFiles> projectFilesList = projectFilesMapper.selectAll(whereMember);
+        for (ProjectFiles projectFiles : projectFilesList) {
+          project.addProjectFilesList(projectFiles);
+        }
+      }
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new Exception(e);
+    }
+    return list;
+  }
+
+  public boolean add(Project project) throws Exception {
+    LOG.info("add({})", project.toString());
+
+    try (SqlSession sqlSession = MyBatisUtil.getSqlSession()) {
+      ProjectMapper projectMapper = sqlSession.getMapper(ProjectMapper.class);
+      projectMapper.insert(project);
+
+      ProjectFilesMapper projectFilesMapper = sqlSession.getMapper(ProjectFilesMapper.class);
+      // add ProjectFiles, when add project, should insert 'ProjectFilesList' to ProjectFiles
+      List<ProjectFiles> list = project.getProjectFilesList();
+      for (ProjectFiles projectFiles : list) {
+        // ProjectId needs to be obtained after the Project is inserted into the database
+        projectFiles.setProjectId(project.getId());
+        projectFilesMapper.insert(projectFiles);
+      }
+
+      sqlSession.commit();
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new Exception(e);
+    }
+    return true;
+  }
+
+  public boolean updateByPrimaryKeySelective(Project project) throws Exception {
+    LOG.info("updateByPrimaryKeySelective({})", project.toString());
+    try (SqlSession sqlSession = MyBatisUtil.getSqlSession()) {
+      ProjectMapper projectMapper = sqlSession.getMapper(ProjectMapper.class);
+      projectMapper.updateByPrimaryKeySelective(project);
+
+      ProjectFilesMapper projectFilesMapper = sqlSession.getMapper(ProjectFilesMapper.class);
+      Map<String, Object> where = new HashMap<>();
+      where.put("projectId", project.getId());
+      // Take two lists of difference
+      List<ProjectFiles> old_projectFiles = projectFilesMapper.selectAll(where);
+      List<String> old_projectFiles_id = new ArrayList<>();
```

Comment:
change `old_projectFiles_id ` to `oldProjectFilesId `

Reply:
Done.